### DISCUSSION TITLE
Python Package & Home Assistant Custom Component

### DISCUSSION THREAD

**Discussion by OliverCullimore** [robbrad/UKBinCollectionData#299] [General]
@robbrad @dp247 What are your thoughts on when we create the first tag that will trigger the first upload of the package to PyPi?

One question would be what version we want to start at?
v1.0.0 or v0.1.0 or similar?

Another question is the version numbering scheme we follow, I assume if we went with semantic versioning it would apply a bit like the below for us:
- Patch (x.y.Z) for bug fixes
- Minor (x.Y.z) for adding a new council
- Major (X.y.z) for breaking changes (changes to the input parameters/output format)

Is there anything we want to do/finish before we release the first version?

**Comment by robbrad** [comment]
I think let’s go with 0.1.0 as a first release and 100% agree with the semantic versioning.

Maybe we keep it as 0.y.z until the home assistant component?

**Reply by OliverCullimore** [reply]
Sounds good, I'll let you do the honors and create a tag for 0.1.0 if we're happy to release a version now.

**Comment by robbrad** [comment]
Started work on the HA comp https://github.com/robbrad/UKBinCollectionData/tree/ha_custom_comp

**Comment by robbrad** [comment]
@OliverCullimore - worked a treat !! https://pypi.org/project/uk-bin-collection/

**Reply by OliverCullimore** [reply]
We'll have to see if you can pip install it and then simply run
`uk_bin_collection [-h] [-p POSTCODE] [-n NUMBER] [-u UPRN] module URL`

If that works we can start with simplifying the install instructions outside of contributing.md

**Reply by atv2016** [reply]
i would be very interested in a HACS component for Homeassistant. How far are we from achieving this ?

**Comment by OliverCullimore** [comment]
@robbrad @dp247 did one of you want to announce the custom component in the Home Assistant thread now we've got automated releases working?
